### PR TITLE
New error raise and unit test for missing objective when using ScipyOptimizer

### PR DESCRIPTION
**Pull request by DKilkenny**

### CODE REVIEW

**Comment by DKilkenny** [comment]
Made addition to pyoptsparse and genetic as well as unit tests for each.

File: openmdao/drivers/tests/test_scipy_optimizer.py [hunk 1]
```diff
@@ -345,6 +345,34 @@ def test_unsupported_equality(self):
...
+
+        with self.assertRaises(Exception) as raises_msg:
+            prob.run_driver()
+
+        exception = raises_msg.exception
+
+        msg = "ScipyOptimizeDriver requires objective to be declared"
+
+        self.assertEqual(exception.args[0], msg)
+
```
Comment (Kenneth-T-Moore) [inline openmdao/drivers/tests/test_scipy_optimizer.py]: Some extra carriage returns in these tests. 2 blank lines should only be at the start of a new class.

Comment (DKilkenny) [reply openmdao/drivers/tests/test_scipy_optimizer.py]: Fixed in latest push

File: openmdao/core/driver.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from six import iteritems, itervalues, string_types
 
 import numpy as np
+import re
```
Comment (naylor-b) [inline openmdao/core/driver.py]: doesn't look like you added any regex here.  Do you need this import?

File: openmdao/drivers/genetic_algorithm_driver.py [hunk 3]
```diff
@@ -221,6 +221,9 @@ def run(self):
         Pm = self.options['Pm']  # if None, it will be calculated in execute_ga()
         Pc = self.options['Pc']
 
+        # Check for missing objective
```
Comment (naylor-b) [inline openmdao/drivers/genetic_algorithm_driver.py]: this comment is unnecessary.  The function name very clearly indicates what's happening.

**Pull request merged**